test/runtests.jl

```diff
@@ -202,18 +219,23 @@ struct B{T, X, Y}
 end
 Setfield.constructor_of(::Type{<: B{T}}) where T = B{T}
 
-@testset "type change during @set" begin
+@testset "type change during @set (custom constructor_of)" begin
     obj = B{1}(2,3)
     obj2 = @set obj.y = :three
     @test obj2 === B{1}(2, :three)
 end
 
-@static if Pkg.installed("StaticArrays") != nothing
-    using StaticArrays
-    obj = StaticArrays.@SMatrix [1 2; 3 4]
-    l = @lens _[2,1]
-    @test get(l, obj) == 3
-    @test_broken set(l, obj, 5) == StaticArrays.@SMatrix [1 2; 5 4]
-    @test_broken setindex(obj, 5, 2, 1) == StaticArrays.@SMatrix [1 2; 5 4]
+if Pkg.installed("StaticArrays") != nothing
+    include("test_staticarrays.jl")
     include("spaceship.jl")
 end
+
+if Pkg.installed("QuickTypes") != nothing
+    @testset "QuickTypes" begin include("test_quicktypes.jl") end
+end
+
+if Pkg.installed("Reconstructables") != nothing
```

Comment:
The `if Pkg.installed("StaticArrays")` was because installing `StaticArrays` on `v0.7` broke CI. I think installing `QuickTypes` is no problem, so you could remove the condition. Same applies for `Reconstructables` / `Kwonly` I guess.

Reply:
It seems CI fails around QuickTypes test.
https://travis-ci.org/jw3126/Setfield.jl/jobs/351898265

It looks like QuickTypes failed in v0.7 (though it's 5 months ago)
https://travis-ci.org/cstjean/QuickTypes.jl/builds/281258251

Anyway, I'll use the conditional only for excluding v0.7 incompatible packages.

Reply:
Maybe we should do 
```julia
if VERSION <v"0.7-"
    Pkg.add("SomePackage")
    using SomePackage
    # do tests
end
```